[WIP] Obsidian Website redesign

psikoi opened the pull request:
### Only the homepage has been styled, this is still WIP, but published for feedback.

psikoi commented:
Update 28/02

- Restyled tags and tags/show/ pages
- Fixed lint errors
- Recolored the loader orange

psikoi commented:
Update 28/02 (again)

- Updated hero background images
- Restyled Blog and Blog List pages
- Changed the bold header fonts from Cocogoose to Lato (Cocogoose doesn't support numbers)

deathbeam commented:
https://www.npmjs.com/package/lato-font i would prefer if you just use existing npm packages instead of putting fonts to repo, if its possible

deathbeam commented:
Also, do you really need to use ids in hero for contribute/download etc? just css class would be preferred, unless you have reason to link it from outside with hash

psikoi commented:
Update 28/02 Pt. 3

- Load lato font with npm
- Fix tag item examine hover

psikoi commented:
Update 28/02 Pt. 4

- Restyle features page

deathbeam commented:
You should also maybe make the builds pass so people can actually view this on netlify :d

deathbeam commented:
Build is still passing, not sure if you are still working on this or not, but I am not going to review this more until the builds are at least passing.

psikoi commented:
I am still working on it, just haven't pushed in a while, will do.

psikoi commented:
Update 11/03

- Reverted download button to a segmented button
- Fixed lint errors

psikoi commented:
Update 12/07

- Rebased
- Restyled Account/Home
- Restyled Account/LootTracker

Can't currently test the grand exchange route, so that is on hold.

psikoi commented:
Update 12/07 pt.2

- Added SearchBar component
- Restyled Account/GrandExchange
- Added tooltip hover on grand exchange records' timestamps.

@deathbeam This needs review, I believe it's finished.

deathbeam commented:
Hmm, I will check it again. And for the failures, maybe it is because of this? https://github.com/runelite/runelite.net/pull/211/files#diff-761a799c5e8d05e357e89ed215584dcfR93

deathbeam commented:
Or at least, why is it there

psikoi commented:
I didn't add that, It came from a rebase on master.

psikoi commented:
But I just checked, master doesn't have it, so idk what happened, I surely didn't write that.

deathbeam commented:
Probably rebase that went wrong :p

psikoi commented:
I'll remove it asap

deathbeam commented:
So I went through this, and both code and design looks very good :ok_hand: 

Only remaining issue is about what we talked in Discord, and that is using `em` instead of `px` font sizes when possible so they will be scaled by browser.

deathbeam commented:
Oh also, it looks like you have to yet to work on the /tags section, right? Or at least atm, it has some weird font :d

deathbeam commented:
And features/ section has black background, I think that is because it leaked from home page change, it should be gray like everywhere else

deathbeam commented on src/routes/features.js:
```diff
@@ -9,8 +9,8 @@ const Features = () => (
-    <section id="features" style={{ background: '#141414' }}>
-      <div class="content-section">
+    <section id="features" style={{ background: '#232323' }}>
+      <div class="content-section" style={{ background: '#232323' }}>
```
this probably should not be hardcoded

deathbeam commented on src/routes/features.js:
```diff
@@ -9,8 +9,8 @@ const Features = () => (
 
-    <section id="features" style={{ background: '#141414' }}>
-      <div class="content-section">
+    <section id="features" style={{ background: '#232323' }}>
```
same here

psikoi commented:
Update 15/07

- Optimize the layout responsiveness for largers displays

psikoi commented:
Update 15/07 pt. 2

- Added navigation toggling

In the current live version of the website, the mobile navigation doesn't close when a link is clicked, the user having to close it manually.

With this change, the menu will toggle onClick

deathbeam commented on src/components/hero.css:
```diff
@@ -108,6 +109,49 @@
+}
+
+/* Large displays */
+@media (min-width: 2600px) {
```
since site is using bootstrap, I would just use same media queries as Bootstrap is using (they even have sass variables for them). See here: https://getbootstrap.com/docs/4.0/layout/overview/

deathbeam commented:
Also, not sure what happened here but it looks wrong:

![](https://i.imgur.com/iAkjwor.png)

psikoi commented:
Update 17/07 

- Fixed navbar styling (margin issues)

deathbeam commented on src/routes/home.js:
```diff
@@ -26,44 +29,93 @@ const Home = ({ commit, release, sessionCount }) => (
+              title="RuneLite Demo Video"
+              width="100%"
+              height="100%"
+              src="https://www.youtube.com/embed/l2vUWMyQ594"
```
this video url should be probably configurable from hero.js

deathbeam commented on src/routes/home.js:
```diff
@@ -26,44 +29,93 @@ const Home = ({ commit, release, sessionCount }) => (
+          <div class="info">
+            <h1>WHAT IS RUNELITE</h1>
+            <p style={{ 'white-space': 'pre-line' }}>
+              Curabitur pretium ut risus non eleifend. Vestibulum ante ipsum
```
This text as well (from hero.js)

psikoi commented:
Update 19/07

- Moved some data to the hero.js "_data" file
- Improved font scaling

deathbeam commented:
Looks good. I still think you shoudl use bootstrap media queries. But, I guess if it works it works, I can clean it up later maybe and use more boostrap stuff.

psikoi commented:
Update 20/07

- Fixed a few mobile inconsistencies, reported by the fellas on discord

deathbeam commented:
Needs rebase

deathbeam commented:
Actually, I also noticed that tooltips are no longer centered for tags page:

https://deploy-preview-211--runelite.netlify.com/tag/show/test,638,638,638,639,640,650,671

gc commented:
![image](https://user-images.githubusercontent.com/30398469/63817550-2c6f4f80-c980-11e9-8fe8-737005cac119.png)

This hover could use better contrast (should it be white?)
It's also off by 1 pixel because of the invisible border + margin.

P.S This redesign looks very very nice! excellent work

gc commented:
The gh-auth callback page needs fixing up: https://deploy-preview-211--runelite.netlify.com/gh-auth-code?code=hello

gc commented:
I also noticed that the item images in the loot tracker (and maybe elsewhere) look distorted:

Distorted version:
![chrome_solmb3AYVO](https://user-images.githubusercontent.com/30398469/64417219-06fae800-d0dc-11e9-8635-4be112209e56.png)

Undistorted (after adding this CSS):
```diff
.drop-wrapper .card-img-top {
    padding: 2px;
+   width: auto;
}
```
![chrome_JQNQl030Lx](https://user-images.githubusercontent.com/30398469/64417279-23972000-d0dc-11e9-9c01-8f2df94bacef.png)

deathbeam commented:
Thanks for testing it @gc . @psikoi are you planning to update the PR?

psikoi commented:
Yes of course, I've just been busy. When are you looking to release it?

gc commented:
@psikoi I'd be down to PR some fixes/changes to your branch if you'd find that helpful. Let me know

psikoi commented:
Update 08/09

- Removed about section (until a video is available)
- Fixed uncentered tag tooltip
- Restyled Github Auth page
- Fixed Loot Tracker item distortion
- Improved styling on hero download menu

Thanks to @gc :)

gc commented:
Few little things with that dropdown (I've included imgur links showing the thing im talking about + possible solutions to try to save time on fixing):

1. dropdown menu is off by 1 pixel to the left, perhaps don't give it any border, and instead give `margin: 1px 0px 0px 0px;`? https://i.imgur.com/MPKaRAm.png

2. The hovered item is off by 1/2 pixels on the right (much more noticeable than the first one). Every `dropdown-item` has those 2 extra pixels but they're hidden because the non-hovered ones have a transparent background, so it seems to be fixable by just giving them a background-color of `#fff;` instead of `transparent`. https://i.imgur.com/CbD0dJQ.png

3. "active" item has a blue bg color, should it be a darker obsidian color? `.dropdown-item.active, .dropdown-item:active`. If this one is changed, the `:hover` should maybe be changed too, its a very transparent blue atm. https://i.imgur.com/OAa5mOo.png

gc commented:
A couple little things I noticed on the https://deploy-preview-211--runelite.netlify.com/account/home page

1. Email can overlap outside of its' container on smaller widths (should it use an ellipsis when it starts to reasonably become too long to fit?) 
![image](https://user-images.githubusercontent.com/30398469/64518268-33f60780-d335-11e9-8cb9-4ac22055de2f.png)

2. Seems like it (`.account-container`) has way too much padding at the bottom/top on mobile/small sizes. I think other pages might also have this issue.

![image](https://user-images.githubusercontent.com/30398469/64518484-9bac5280-d335-11e9-8519-14f276774bd8.png)

psikoi commented:
🎉 deployed!
